new-log-viewer/src/utils/actions.ts

```diff
@@ -67,42 +61,32 @@ const EDITOR_ACTIONS : ActionType[] = [
         keyBindings: [monaco.KeyMod.CtrlCmd | monaco.KeyCode.KeyI],
     },
 ];
-/* eslint-enable sort-keys */
 
-/**
- * Handles an action based on the given action name, log event number, and total number of events.
- *
- * @param actionName
- * @param logEventNum
- * @param numEvents
- */
-const handleAction = (actionName: ACTION_NAME, logEventNum: number, numEvents: number) => {
-    const pageSize = getConfig(CONFIG_KEY.PAGE_SIZE);
-    switch (actionName) {
-        case ACTION_NAME.FIRST_PAGE:
-            updateWindowUrlHashParams({logEventNum: 1});
-            break;
-        case ACTION_NAME.PREV_PAGE:
-            updateWindowUrlHashParams({
-                logEventNum: getLastItemNumInPrevChunk(logEventNum, pageSize),
-            });
-            break;
-        case ACTION_NAME.NEXT_PAGE:
-            updateWindowUrlHashParams({
-                logEventNum: getFirstItemNumInNextChunk(logEventNum, pageSize),
-            });
-            break;
-        case ACTION_NAME.LAST_PAGE:
-            updateWindowUrlHashParams({logEventNum: numEvents});
-            break;
-        default:
-            break;
-    }
+type NavigationActionsMap = {
+    [ACTION_NAME.SPECIFIC_PAGE]: {
+        specificPageNum: number,
+    },
+    [ACTION_NAME.FIRST_PAGE]: null
+    ,
+    [ACTION_NAME.PREV_PAGE]: null
+    ,
+    [ACTION_NAME.NEXT_PAGE]: null
+    ,
+    [ACTION_NAME.LAST_PAGE]: null
 };
 
+// Some navigation actions supply code and args fields, and others only code field.
```

Comment:
Why not make `args` always exist? You're already setting it to `null` in `NavigationActionsMap`, right?

Reply:
Yeah, I think I prefer a simplified type rather than simplified call sites.